process.go

```diff
@@ -407,3 +410,14 @@ func (p *Process) createTasks() (ch chan *Task) {
 	}()
 	return ch
 }
+
+type taskQueue []*Task
+
+// NextSignal allows us to wait for the next task to be done if it's available.
+// Otherwise, nil is returned since nil channels always block.
+func (tq taskQueue) NextSignal() chan int {
```

Comment:
This can also be done without wrapping slice in another type. That would require the earlier code to be responsible for updating/`nil`ing the next `Done` channel as messages are processed.

Reply:
Wrapping in a new type is all fine IMO. Only I found "NextSignal" to be a bit unclear, and had me look up the definition to understand. What do you think about e.g. `NextTaskDone()`?

Reply:
```suggestion
func (tq taskQueue) NextTaskDone() chan int {
```
That name works for me. I think by using this new "suggestion" feature, we can make/accept the change from here.